autoload/ale/lsp.vim

```diff
@@ -260,8 +261,12 @@ function! s:UpdateCapabilities(conn, capabilities) abort
         let a:conn.capabilities.symbol_search = 1
     endif
 
-    if type(get(a:capabilities, 'workspaceSymbolProvider')) is v:t_dict
-        let a:conn.capabilities.symbol_search = 1
+    if has_key(a:capabilities, 'textDocumentSync')
+        if has_key(a:capabilities.textDocumentSync, 'save')
+            if get(a:capabilities.textDocumentSync.save, 'includeText') is v:true
+                let a:conn.capabilities.includeText = 1
+            endif
+        endif
```

Comment:
It seems this chunk could use a `try` block like this:

```vimscript
    try
        if get(a:capabilities.textDocumentSync.save, 'includeText') is v:true
            let a:conn.capabilities.includeText = 1
        endif
    endtry
```

But I'm not really familiar with vimscript.

Reply:
Why do you want a `try` here?

If you are worried about `'includeText'` not existing, you could add a third argument to `get()` that returns a default value in that case. However, since `get()` by default returns `0` if the key is not found, and `0` is **not** `v:true`, this isn't even necessary here.

Reply:
I'm worried about `textDocumentSync` and `save` not existing due to a buggy LSP server implementation, which is why I thought the `try` block could help.

Reply:
> I'm worried about `textDocumentSync` and `save` not existing

This isn't possible, since you check `has_key(a:capabilities, 'textDocumentSync')` beforehand.

Reply:
I think I didn't explain myself correctly. What I meant to say initially was that it seemed that this chunk could use a `try` block and that those `has_key` calls could be dropped for readability. The `try` block is just for the possibility of `textDocumentSync` and `save` not existing.

Essentially turning this:
```
    if has_key(a:capabilities, 'textDocumentSync')
        if has_key(a:capabilities.textDocumentSync, 'save')
            if get(a:capabilities.textDocumentSync.save, 'includeText') is v:true
                let a:conn.capabilities.includeText = 1
            endif
        endif
    endif
```
Into this:
```
    try
        if get(a:capabilities.textDocumentSync.save, 'includeText') is v:true
            let a:conn.capabilities.includeText = 1 endif endtry
        endif
    endtry
```

Reply:
Not expert in LSP but from what I read in the documentation the "textDocumentSync.save" property can be either a boolean or SaveOptions type. So it must be tested first as a boolean and then as a dict. The same treatment can be seen on the other capabilities where they are tested for v:true first and then for v:t_dict.

Something like this should do and is more readable than using try/catch to handle program logic:

```
if type(get(a:capabilities, 'textDocumentSync')) is v:t_dict
  let l:save = get(a:capabilities.textDocumentSync, 'save')

  if l:save is v:true
    let a:conn.capabilities.includeText = 1
  endif

  if type(l:save) is v:t_dict
    let a:conn.capabilities.includeText = get(l:save, 'includeText', v:false)
  endif
endif
```

Another advantes of this implementation is that in future other attributes in the textDocumentSync dictionary can be easily added.

Note I have not tested this.

Reply:
Thanks for the suggestion. I've implemented this as well.